### DISCUSSION TITLE
Case study presenting a view controller from a detached view controller freezes

### DISCUSSION THREAD

**Discussion by teddy-bersentes** [pointfreeco/swift-composable-architecture#1409] [General]
### Description

As I was experimenting with the SwiftUI dismissal swipe gesture (the gesture where you swipe from left to right when going back to a screen), I found that the case study "03-Navigation-Sheet-LoadThenPresent" freezes when the popup triggers. I initially found this bug on the `protocol` branch, but I am able to consistently reproduce this on the most recent `main` branch. 

The error in the console in the event of the screen freezing prints out:
`
2022-09-23 17:15:35.976082-0400 SwiftUICaseStudies[2428:123121] [Presentation] Presenting view controller <_TtGC7SwiftUI29PresentationHostingControllerVS_7AnyView_: 0x13160fb70> from detached view controller <_TtGC7SwiftUI19UIHostingControllerGVS_15ModifiedContentVS_7AnyViewVS_24NavigationColumnModifier__: 0x131609a00> is discouraged.
`

### Checklist

- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues).

### Expected behavior

The dismissal gesture cancels and the screen behind the popup returns to its initial state. It could also continue the gesture, where the user would navigate to the previous screen, but the popup would dismiss.

### Actual behavior

You keep control of the navigation gesture but with the popup sheet still, on top of that page. You can let go when the page is near the right side of the screen, resulting in a dismissal of that page and keeping the popup on your screen. You can also let go when the page is near the left, resulting in the screen freezing indefinitely.

### Steps to reproduce

1. Go to the "Sheets: Load data then present" case study on the SwiftUI CaseStudies app
2. Press on the "Load optional count" button
3. Before the loading is complete, begin the swipe back gesture from the left but do not release
4. As you continue to hold after the popup appears, you can release the gesture near the left or right

https://user-images.githubusercontent.com/47639017/192058259-c8153b83-fe00-468d-973c-1bedbcf94a4d.mp4

### The Composable Architecture version information

0.40.2

### Destination operating system

iOS 15.5

### Xcode version information

Version 14.0 (14A309)

### Swift Compiler version information

```shell
swift-driver version: 1.62.8 Apple Swift version 5.7 (swiftlang-5.7.0.127.4 clang-1400.0.29.50)
Target: arm64-apple-macosx12.0
```

**Comment by mbrandonw** [comment]
This is an issue with vanilla SwiftUI too as the following demonstrates:

```swift
struct ContentView: View {
  @State var isPresented = false

  var body: some View {
    NavigationStack {
      NavigationLink("Drill down") {
        Button("Present") {
          Task {
            try await Task.sleep(nanoseconds: NSEC_PER_SEC * 2)
            self.isPresented = true
          }
        }
        .sheet(isPresented: self.$isPresented) {
          Text("Hi")
        }
      }
    }
  }
}
```

I don't know of a workaround (in TCA or in vanilla SwiftUI) as there doesn't seem to be anyway to be notified if one is in the middle of a drag gesture so that you can cancel any inflight work.

I'm going to move this to a discussion since this is not an issue with the library.

**Comment by teddy-bersentes** [comment]
Yeah, I assumed this was an issue with vanilla SwiftUI.

I suppose a workaround to prevent this would be to disable the swipe-back gesture during the loading state. I am still pretty new to TCA, so for anyone facing this issue, [this StackOverflow answer](https://stackoverflow.com/a/69380125/13737173) might be a good starting point.
